clients/python/test/e2e/test_files_api.py

```diff
@@ -34,106 +32,80 @@ def _hash_file(file: Path) -> str:
+        retval=True,
+    )
+    assert (
+        large_server_file.server_file.id == uploaded_file.id
+    ), "could not detect that file was already on server"
+    assert (
+        max_diff(upload_ram_usage_in_mb) < _allowed_ram_usage_in_mb
+    ), f"Used more than {_allowed_ram_usage_in_mb=} to upload file of size {large_server_file.local_file.stat().st_size=}"
+    download_ram_usage_in_mb, downloaded_file = memory_usage(
+        (
+            files_api.download_file,
+            (uploaded_file.id,),
+            {"destination_folder": tmp_path},
+        ),  # type: ignore
+        retval=True,
+    )
+    assert Path(downloaded_file).parent == tmp_path
+    assert (
+        max_diff(download_ram_usage_in_mb) < _allowed_ram_usage_in_mb
+    ), f"Used more than {_allowed_ram_usage_in_mb=} to download file of size {Path(downloaded_file).stat().st_size=}"
+    assert _hash_file(Path(downloaded_file)) == _hash_file(large_server_file.local_file)
 
 
 @skip_if_osparc_version(at_least=Version("0.8.0"))
 @pytest.mark.parametrize("use_checksum", [True, False])
 @pytest.mark.parametrize("use_id", [True, False])
 def test_search_files(
-    create_tmp_file: Callable[[ByteSize], Path],
-    api_client: osparc.ApiClient,
+    large_server_file: Callable[[ByteSize], Path],
+    files_api: osparc.FilesApi,
     use_checksum: bool,
     use_id: bool,
+    faker: Faker,
 ) -> None:
-    tmp_file = create_tmp_file(ByteSize(1 * _GB))
-    checksum: str = _hash_file(tmp_file)
-    results: osparc.PaginationGenerator
-    files_api: osparc.FilesApi = osparc.FilesApi(api_client=api_client)
-    try:
-        results = files_api._search_files(sha256_checksum=checksum)
-        assert len(results) == 0, "Found file which shouldn't be there"
-
-        uploaded_file: osparc.File = files_api.upload_file(tmp_file)
-        assert checksum == uploaded_file.checksum
-
-        results = files_api._search_files(
-            file_id=uploaded_file.id if use_id else None,
-            sha256_checksum=uploaded_file.checksum if use_checksum else None,
-        )
-        assert len(results) == 1, "Could not find file after it had been uploaded"
-
-        files_api.delete_file(uploaded_file.id)
-        results = files_api._search_files(
-            file_id=uploaded_file.id if use_id else None,
-            sha256_checksum=uploaded_file.checksum if use_checksum else None,
-        )
-        assert len(results) == 0, "Could find file on server after it had been deleted"
-
-    except Exception:
-        # clean up in case of failure
-        results = files_api._search_files(sha256_checksum=checksum)
-        for file in results:
-            files_api.delete_file(file.id)
+    results: osparc.PaginationGenerator = files_api._search_files(
+        sha256_checksum=f"{faker.sha256()}"
+    )
+    assert len(results) == 0, "Found file which shouldn't be there"
+
+    results = files_api._search_files(
```

Comment:
Check https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes to define the returing value

Reply:
This is a very good point. I will do it, but in another PR to not pollute this one